src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -860,7 +807,7 @@ private ExtendedSequenceNumber convertToCheckpoint(InitialPositionInStreamExtend
     /** Helper class to compare leases based on starting sequence number of the corresponding shards.
      *
      */
-    private static class StartingSequenceNumberAndShardIdBasedComparator implements Comparator<KinesisClientLease>,
+    static class StartingSequenceNumberAndShardIdBasedComparator implements Comparator<KinesisClientLease>,
```

Comment:
Is the implementation of this missing in the diff? I think you might have meant to include this.

Reply:
This was already implemented, I just changed the privacy level. But moving synchonizers back into inner classes will remove this in the next revision.